code/datums/abilities/flock/flockmind.dm

```diff
@@ -52,15 +77,44 @@
 	name = "Spawn Rift"
 	desc = "Spawn an rift where you are, and from there, begin."
 	icon_state = "spawn_egg"
-	targeted = 0
+	targeted = FALSE
 	cooldown = 0
 
 /datum/targetable/flockmindAbility/spawnEgg/cast(atom/target)
 	if(..())
-		return 1
+		return TRUE
+
 	var/mob/living/intangible/flock/flockmind/F = holder.owner
-	if(F)
-		F.spawnEgg()
+
+	var/turf/T = get_turf(F)
+
+	if (istype(T, /turf/space/) || istype(T.loc, /area/station/solar) || istype(T.loc, /area/station/mining/magnet))
+		boutput(F, "<span class='alert'>Space and exposed areas are unsuitable for rift placement!</span>")
+		return TRUE
```

Comment:
Minor gripe, very unimportant- can this go after the `if (!isadmin)`? Thinking about the very niche case where an admin wants to plant on the mining magnet as an event or something, or spawn on a built zone which is Technically space. 

It's fine if spawning outside fundamentally breaks the mode but frankly if an admin wants to spawn somewhere stupid, let em

Reply:
AFAIK this is copy-pasted from blob verbatim. Will double-check

Reply:
https://github.com/stonepillars/goon-flock/pull/320